src/nvim/edit.c

```diff
@@ -2738,124 +2792,231 @@ static void trigger_complete_changed_event(int cur)
   tv_dict_clear(v_event);
 }
 
-/// Show the popup menu for the list of matches.
-/// Also adjusts "compl_shown_match" to an entry that is actually displayed.
-void ins_compl_show_pum(void)
+// comparator function for use in qsort.
+static int compare_scores(const void *a, const void *b)
+{
+  double val = ((compl_score_T *)b)->score - ((compl_score_T *)a)->score;
+  if (val < 0) {
+    return -1;
+  } else if (val > 0) {
+      return 1;
+  } else {
+    // qsort is not stable: we don't want to flip two items that have
+    // the same score.
+    return ((compl_score_T *)a)->order - ((compl_score_T *)b)->order;
+  }
+}
+
+// sort the linked list of completion items based on the scores
+// in the scores array.
+static void sort_completions(int num_items, compl_score_T *scores)
```

Comment:
use common prefixes. `ins_compl_` looks like what other related functions use.
```suggestion
static void ins_compl_sort(int num_items, compl_score_T *scores)
```